pkg/cloud/services/compute/instance.go

```diff
@@ -470,12 +471,19 @@ func (s *Service) getOrCreatePort(eventObject runtime.Object, clusterName string
 		SecurityGroups:      securityGroups,
 		AllowedAddressPairs: []ports.AddressPair{},
 	}
-
-	for _, ap := range portOpts.AllowedAddressPairs {
-		createOpts.AllowedAddressPairs = append(createOpts.AllowedAddressPairs, ports.AddressPair{
-			IPAddress:  ap.IPAddress,
-			MACAddress: ap.MACAddress,
-		})
+	// do not add allowed-address-pairs
+	if portOpts.DisablePortLevelSecurity == false {
+		for _, ap := range portOpts.AllowedAddressPairs {
+			createOpts.AllowedAddressPairs = append(createOpts.AllowedAddressPairs, ports.AddressPair{
+				IPAddress:  ap.IPAddress,
+				MACAddress: ap.MACAddress,
+			})
+		}
+	}
+	// do not add security groups
+	if portOpts.DisablePortLevelSecurity == true {
+		createOpts.SecurityGroups = &[]string{}
+		createOpts.AllowedAddressPairs = []ports.AddressPair{}
```

Comment:
This line isn't needed. `AllowedAddressPairs` is already set to this value when your boolean is true.

Reply:
> Change looks good. Have you been able to test this in our CityCloud environment?

Yes, it is tested.